python/activator/middleware_interface.py

```diff
@@ -1261,6 +1275,46 @@ def _chain_exports(self, output_chain: str, output_runs: collections.abc.Iterabl
             with self.central_butler.transaction():
                 _prepend_collection(self.central_butler, output_chain, output_runs)
 
+    def _query_datasets_by_storage_class(self, butler, exposure_ids, collections, storage_class):
+        """Identify all datasets with a particular storage class, regardless of
+        dataset type.
+
+        Parameters
+        ----------
+        butler : `lsst.daf.butler.Butler`
+            The Butler in which to query for datasets.
+        exposure_ids : `set` [`int`]
+            Exposure IDs for which to return datasets.
+        collections : iterable [`str`]
+            The collections in which to query for datasets.
+        storage_class : `str`
+            The name of the storage class by which to query.
+
+        Yields
+        ------
+        dataset : `lsst.daf.butler.DatasetRef`
+            A dataset in ``collections`` of type ``storage_class``. Guaranteed
+            to include values for implied dimensions, but need not include
+            dimension records. The order in which datasets are returned
+            is undefined.
+        """
+        matching_types = {dtype for dtype in butler.registry.queryDatasetTypes(...)
```

Comment:
Would there be any value in using an ordered dict here instead?

Reply:
I don't see how. I assume the dataset types would be the keys, but what would the values be?

Reply:
Sorry, I incorrectly read the {.. as defining a dict, and I was looking for ways to make the order more consistent. It is not important.